### PR TITLE
[amplipi] Add discovery and PA support

### PR DESCRIPTION
**Pull request by kaikreuzer**
This PR adds the possibility to find the AmpliPi on the local network through mDNS.

Additionally, it now acts as an audio sink, so that audio streams can be sent to it, which are then announced as a PA on all available zones (either with current volume or a specified volume level). This is a great feature for doing announcements in every room, such as doorbell, telephone call notifications etc.

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
@linknum23 FYI!

File: bundles/org.openhab.binding.amplipi/src/main/java/org/openhab/binding/amplipi/internal/audio/PAAudioSink.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    private static final Set<AudioFormat> SUPPORTED_AUDIO_FORMATS = Set.of(AudioFormat.MP3, AudioFormat.WAV);
+    private static final Set<Class<? extends AudioStream>> SUPPORTED_AUDIO_STREAMS = Set
+            .of(FixedLengthAudioStream.class, URLAudioStream.class);
+
+    private @Nullable AmpliPiHandler handler;
+
+    private @Nullable PercentType volume;
+
+    public PAAudioSink() {
+    }
```
Comment (wborn) [inline bundles/org.openhab.binding.amplipi/src/main/java/org/openhab/binding/amplipi/internal/audio/PAAudioSink.java:55]: Isn't it possible to use the default constructor here and remove this default constructor?

**Review by wborn — COMMENTED** [review]
Thanks for adding this nice enhancement!
It looked like a lot of code but most of it is generated. :wink: 
I only have one small remark and also a question.

File: bundles/org.openhab.binding.amplipi/src/main/java/org/openhab/binding/amplipi/internal/audio/PAAudioSink.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/org.openhab.binding.amplipi/src/main/java/org/openhab/binding/amplipi/internal/audio/PAAudioSink.java]: Indeed, not sure why I had it there...

**Comment by kaikreuzer** [comment]
Thanks @wborn, your remarks are addressed now!

**Review by wborn — APPROVED** [review]
Thank you!

**Pull request merged**